Add minimal docs on how to install gazelle_cc using workspaces

WojciechMazur opened the pull request:
Resolves #39

jayconrod commented on README.md:
```diff
@@ -39,6 +38,12 @@ gazelle(
 
+### Using WORKSPACE
+
+The `gazelle_cc` is build and distributed using using Bazel modules. It is possible to use it using legacy WORKSPACE defintions, but it is neither tested nor encouraged.
```
This is a little strong. Bazel is removing support for `WORKSPACE` files in 9.0 later this year, but we should still support `WORKSPACE` in at least 8.0. A lot of people will be stuck with `WORKSPACE` for a while.

jayconrod commented on docs/workspace_setup.md:
```diff
@@ -0,0 +1,71 @@
+This Gazelle extensions uses by default the new naming convention for external dependencies using bzlmod.
+You might be required to use `repo_mapping` to adjust dependenices installed using WORKSPACE under different naming convention.
+
+<!-- TODO Replace with http_archive when first release is available -->
```
Let's tag a version with a downloadable archive and use `http_archive` here. I don't think we should encourage `git_repository` with a branch at all, since it leads to non-deterministic builds.

WojciechMazur commented:
I've changed the wording for WORKSPACE support, added `example/workspace` and used the tagged release in the docs instead of git_repository. Should be ready for another review round
